src/main/java/org/fcrepo/importexport/importer/Importer.java

```diff
@@ -89,49 +99,84 @@ private FcrepoClient client() {
      */
     public void run() {
         logger.info("Running importer...");
-        importDirectory(config.getDescriptionDirectory());
+        final File importContainerMetadataFile = TransferProcess.fileForContainer(config.getResource(),
+                config.getDescriptionDirectory(), config.getRdfExtension());
+        importContainerDirectory = TransferProcess.directoryForContainer(config.getResource(),
+                config.getDescriptionDirectory());
+        if (!importContainerMetadataFile.exists()) {
+            logger.debug("No container exists in the metadata directory {} for the requested resource {},"
+                    + " importing all contained resources instead.", importContainerMetadataFile.getPath(),
+                    config.getResource());
+        } else {
+            importFile(importContainerMetadataFile);
+        }
+        importDirectory(importContainerDirectory);
     }
 
     private void importDirectory(final File dir) {
-        for (final File f : dir.listFiles()) {
-            if (f.isDirectory()) {
-                importDirectory(f);
-            } else if (f.isFile()) {
-                importFile(f);
-            }
+        for (final File f : dir.listFiles((f) -> {
```

Comment:
I thought we reset Checkstyle not to require pointless and confusing line breaks in lambdas?